src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/BucketHelper.scala

```diff
@@ -102,11 +131,11 @@ class BucketHelper(dataprocConfig: DataprocConfig,
     }
   }
 
-  private def userEntity(email: WorkbenchEmail) = {
-    GcsEntity(email, User)
-  }
+  private def userEntity(email: WorkbenchEmail) = EmailGcsEntity(User, email)
+  private def groupEntity(email: WorkbenchEmail) = EmailGcsEntity(Group, email)
+}
 
-  private def groupEntity(email: WorkbenchEmail) = {
-    GcsEntity(email, Group)
-  }
+object BucketHelper {
+  case class NoGoogleProjectNumberException(googleProject: GoogleProject)
+    extends LeoException(s"Project number could not be found for Google project $googleProject")
```

Comment:
I wonder if this should return 404 instead of 500 (the default). This error message will be serialized and returned to users.